w3lib/url.py

```diff
@@ -68,18 +68,34 @@ def safe_url_string(url, encoding='utf8', path_encoding='utf8'):
 
     # quote() in Python2 return type follows input type;
     # quote() in Python3 always returns Unicode (native str)
-    return urlunsplit((
-        to_native_str(parts.scheme),
-        to_native_str(netloc).rstrip(':'),
-
-        # default encoding for path component SHOULD be UTF-8
-        quote(to_bytes(parts.path, path_encoding), _safe_chars),
-
-        # encoding of query and fragment follows page encoding
-        # or form-charset (if known and passed)
-        quote(to_bytes(parts.query, encoding), _safe_chars),
-        quote(to_bytes(parts.fragment, encoding), _safe_chars),
-    ))
+    
+    if bool(escape_URL):
+        return urlunsplit((
+            to_native_str(parts.scheme),
+            to_native_str(netloc).rstrip(':'),
+
+            # default encoding for path component SHOULD be UTF-8
+            to_native_str(parts.path),
+
+            # encoding of query and fragment follows page encoding
+            # or form-charset (if known and passed)
+            quote(to_bytes(parts.query, encoding), _safe_chars),
+            quote(to_bytes(parts.fragment, encoding), _safe_chars),
+        ))
+      
+    else:
```

Comment:
It would be great to structure this code without duplication - if we want to change something, we'd like to do it in one place.